### DISCUSSION TITLE
XOAUTH2 SMTP/IMAP authentication

### DISCUSSION THREAD

**Discussion by ghost** [curl/curl#8584] [Ideas]
It would be great if curl supports XOAUTH2 (not to be confused with OAUTH bearer) for SMTP(S)/IMAP(S) authentication

 

And most importantly Gmail will no longer accept user and password only authentication after May 30 2022 and will accept OAuth 2.0 tokens only

 ![Screenshot_20220311-080951.png](https://user-images.githubusercontent.com/81270448/157904848-eccefcaf-4338-451a-9245-8772a85e11a2.png)

Information about OAuth 2.0 and how to get such tokens https://developers.google.com/identity/protocols/oauth2

Thunderbird's OAuth source codes would be a great example to learn from

[Unfortunately none of the existing implementation use C code so that will be a challenge]

**Comment by ghost** [comment]
@bagder Are you aware of this change made by major email provider and organisations?

**Reply by bagder** [reply]
Yes, I can read what you post.

**Comment by ghost** [comment]
Unfortunately, as discussed yesterday on IRC, there is no simple way to hack OAuth2

You need to pay Google a huge amount of money for evaluation and auditing for off device accounts

and it requires Google's approval and evaluation for all through the process

At the moment app passwords would be the only way..........
